data/filter/BaseFilterFieldSpec.ts

```diff
@@ -74,7 +75,9 @@ export abstract class BaseFilterFieldSpec extends HoistBase {
         this.forceSelection = forceSelection ?? false;
         this.values = values ?? (this.isBoolFieldType ? [true, false] : null);
         this.hasExplicitValues = !isEmpty(this.values);
-        this.enableValues = this.hasExplicitValues || (enableValues ?? this.isEnumerableByDefault);
+        this.enableValues =
+            !this.isMismatchedDateFieldType &&
+            (this.hasExplicitValues || (enableValues ?? this.isEnumerableByDefault));
```

Comment:
I believe we said we don't want to enable the Values tab in this situation, right? It could be the dev's responsibility to set enableValues to false on the fieldSpec, but I think the default behavior of not showing it for dates makes sense.

Reply:
It would be great to `enableValues` if we could find a clean way to restrict the values to the unique local dates in the dataset, rather than the full date timestamp. If this proves too complex however perhaps its not worth it.